src/Illuminate/Filesystem/FilesystemManager.php

```diff
@@ -140,13 +143,7 @@ protected function resolve($name)
      */
     protected function callCustomCreator(array $config)
     {
-        $driver = $this->customCreators[$config['driver']]($this->app, $config);
-
-        if ($driver instanceof FilesystemInterface) {
-            return $this->adapt($driver);
-        }
-
-        return $driver;
+        return $this->customCreators[$config['driver']]($this->app, $config);
```

Comment:
These changes require custom creators to return an instance of `Illuminate\Contracts\Filesystem\Filesystem` rather than an instance of `League\Flysystem\FilesystemInterface`. The reason for that is that creating a `FilesystemAdapter` instance is now much more complex then what the previous `adapt` method did (see a bit further below).